Add subsection on possible attack on Privacy Pass (#27)
phnx-im/docs: src/threat_model/qualitative/metadata.md

## Privacy Pass

[Privacy Pass](https://datatracker.ietf.org/doc/draft-ietf-privacypass-batched-tokens/) is used to rate-limit individual users without violating their metadata privacy.

The protocol flow is as follows:

- A registered client requests a number of tokens from the AS.
- The AS checks the client's token allowance and issues the requested tokens if the client has not exceeded their allowance. The Privacy Pass protocol ensures that the tokens are not linked to the client's identity.
- Whenever the client needs to interact with an endpoint that is rate-limited using PrivacyPass, the client includes a token as part of its request.

> - The endpoint verifies that the token is valid and has not yet been spent. If the token is valid, the endpoint processes the request and marks the token as spent.

Reviewer: ```suggestion
- The endpoint verifies that the token has previously been issued by the AS and has not yet been spent. If both criteria are met, the endpoint marks the token as spent and continues with processing the request.
```

> To create and verify tokens the AS uses a pair of public/private keys. As long as the AS uses the same keypair for all clients, it cannot correlate the tokens with the clients' identities. However, a malicious AS could use distinct keypairs for each client and thus correlate the tokens with the clients' identities. See [Section 6.2 of RFC 9576](https://www.rfc-editor.org/rfc/rfc9576.html#section-6.2) for more information on this threat.

Reviewer: ```suggestion
In the issuance and redemption phase of tokens the AS uses a pair of public/private keys. As long as the AS uses the same keypair for all clients, it cannot correlate the tokens with the clients' identities. However, a malicious AS could use distinct keypairs for each client and thus correlate the tokens with the clients' identities. See [Section 6.2 of RFC 9576](https://www.rfc-editor.org/rfc/rfc9576.html#section-6.2) for more information on this threat.
```

> The Phoenix homeserver protocol currently does not mitigate this attack. In the future, AS public/private keys will either be gossiped between clients as part of their group state, or a key transparency-like mechanism will be used to allow clients to verifiably track Privacy Pass key material used by the AS.

Reviewer: ```suggestion
The Phoenix homeserver protocol currently does not mitigate this attack. In the future, AS public/private keys will either be gossiped between clients as part of their group state, or a key transparency-like mechanism will be used to allow clients to verifiably track Privacy Pass key material used by the AS.
For now, clients can however still download the public key anonymously at any given point in time, making it harder for the server to issue a specific key per client.
```

## Traffic analysis

Traffic analysis, i.e. the analysis of data traffic patterns, is not considered part of the threat model. While the use of network metadata can allow an active observer (be it on the server or just the network) to infer social graphs and other sensitive information, the Phoenix homeserver protocol does not aim to protect against such attacks. Instead, common countermeasures such as onion routing or the use of mixnets can be used in conjunction with the Phoenix homeserver protocol to mitigate such attacks. However, to the extent possible, the protocol avoids behaviours obvious traffic patterns such as batched queries, messages at regular intervals, etc.

## Privacy Pass

> The [Privacy Passbatched tokens issuance protocol](https://datatracker.ietf.org/doc/draft-ietf-privacypass-batched-tokens/) is used to rate-limit individual users without violating their metadata privacy.

Reviewer: ```suggestion
The [Privacy Pass batched tokens issuance protocol](https://datatracker.ietf.org/doc/draft-ietf-privacypass-batched-tokens/) is used to rate-limit individual users without violating their metadata privacy.
```
